### PR TITLE
subscription: allow dates in ISO 8601 format

### PR DESCRIPTION
**Pull request by ptoscano**
Right now, the date returned by D-Bus method calls of subscription-manager have a locale-specific format, which can be hard to
parse. A planned change is to return all the dates in ISO 8601 format: https://github.com/candlepin/subscription-manager/pull/2551

Hence, make the subscription module able to cope with the future value: first try to parse the date as ISO 8601, and if it does not work try to parse as it is done today. Add small unit testing data to check that both ISO 8601 and the locale-specific formats work.

### CODE REVIEW

**Review by VladimirSlavik — APPROVED** [review]
Looks good to me. Thank you!

**Review by poncovka — APPROVED** [review]
Looks good to me. Thanks!

**Review by M4rtinK — APPROVED** [review]
Looks good to me as well! :)

Thanks a lot for fixing the date formatting, that should make it much more robust going forward! :)

I'll wait for the subscription manager side fix to land so I can test this PR manually and will then assure it finds it's way to the relevant RHEL branches. :)

**Review by jkonecny12 — APPROVED** [review]
Looks great. Thanks a lot!

**Comment by ptoscano** [comment]
Thanks for the reviews and approvals!

The PR for subscription-manager was approved, and I'm proposing https://github.com/candlepin/subscription-manager/pull/2570 for 1.28 (i.e. future RHEL).

**Comment by M4rtinK** [comment]
Unfortunately I was not able to get the updated subscription manager  injected into Fedora Rawhide boot iso via updates image going. :P Oh well, the code still looks fine so lets merge this.

We can always re-test on RHEL before merging the backport PR.

**Comment by M4rtinK** [comment]
Port to RHEL 8: #3329 
Port to RHEL 9: #3330

**Pull request merged**
